docusaurus/docs/React/guides/sdk-state-management.mdx

```diff
@@ -0,0 +1,146 @@
+:::
+
+Channel as a prop example:
+
+```tsx
+import { useState, useEffect } from 'react';
+import { Channel, useChatContext } from 'stream-chat-react';
+
+export const ChannelWrapper = ({ channelId, channelType = 'messaging', children }) => {
+  const [activeChannel, setActiveChannel] = useState(undefined);
+  const { client } = useChatContext();
+
+  useEffect(() => {
+    if (!channelId) return;
+
+    const channel = client.channel(channelId, channelType);
+
+    setActiveChannel(channel);
+  }, [channelId, channelType]);
+
+  return <Channel channel={activeChannel}>{children}</Channel>;
+};
+```
+
+Example with `setActiveChannel` function:
+
+```tsx
+import { useEffect } from 'react';
+import { useCreateChatClient, useChatContext, Chat, Channel } from 'stream-chat-react';
+
+const ActiveChannelSetter = ({ channelId, channelType }) => {
+  const { client, setActiveChannel } = useChatContext();
+
+  useEffect(() => {
+    const channel = client.channel(channelId, channelType);
+    setActiveChannel(channel);
+  }, [channelType, channelId]);
+
+  return null;
+};
+
+const App = () => {
+  const client = useCreateChatClient(userData);
+
+  if (!client) return <div>Loading...</div>;
+
+  return (
+    <Chat client={client}>
+      <ActiveChannelSetter channelId='random' channelType='messaging' />
+      <Channel>{/** ...other components... */}</Channel>
+    </Chat>
+  );
+};
+```
+
+Afterwards - to access currently active channel you can use `useChatContext` hook as well from any component (or custom hook) which is either direct or indirect child of a Chat component:
+
+```tsx
+import { useChatContext } from 'stream-chat-react';
+
+export const useActiveChannel = () => {
+  const { channel } = useChatContext();
+
+  return channel;
+};
+```
+
+Currently active [channel state](https://github.com/GetStream/stream-chat-react/blob/master/src/context/ChannelStateContext.tsx#L36-L56) can be accessed through the [`ChannelStateContext`](../components/contexts/channel-state-context.mdx) with the help of the `useChannelStateContext` hook - meaning any component which is either direct or indirect child of the `Channel` component can access such state.
+
+The example bellow shows how to reach `members` property coming from the channel state:
+
+```tsx
+import { Channel, useChannelStateContext } from 'stream-chat-react';
+
+const MembersCount = () => {
+  const { members } = useChannelStateContext();
+
+  return <div>{Object.keys(members).length}</div>;
+};
+
+const ChannelWrapper = () => (
+  <Channel>
+    <MembersCount />
+  </Channel>
+);
+```
+
+## Channel List State
+
+[`ChannelList`](../components/core-components/channel-list.mdx) component is a standalone component which (unsurprisingly) holds and manages list of channels. You can access loaded `channels` from [`ChannelListContext`](../components/contexts/channel-list-context.mdx) with the help of `useChannelListContext` hook.
```

Comment:
Let's specifically mention that this context is only accessible from the `ChannelList` subcomponents (Preview, List, etc)